Auth0/OAuth2Grant.swift

```diff
@@ -30,9 +31,34 @@ protocol OAuth2Grant {
 
 struct ImplicitGrant: OAuth2Grant {
 
-    let defaults: [String : String] = ["response_type": "token"]
+    let defaults: [String : String]
+    let response: [ResponseOptions]
+
+    init(response: [ResponseOptions] = [.token], nonce: String? = nil) {
+        self.response = response
+        if let nonce = nonce {
+            self.defaults = ["nonce" : nonce]
+        } else {
+            self.defaults = [:]
+        }
+    }
 
     func credentials(from values: [String : String], callback: @escaping (Result<Credentials>) -> ()) {
+        // id token reponse expectations and JWT validation, nonce validation
+        if response.contains(.id_token) {
```

Comment:
Multiple *responseType* are processed in this `func`, `guard` would exit if the *responseType* was not `.id_token`, even `.id_token` would then fail in the next proposed guard as it wouldn't contain `.token`
This applies to all my usage of `if` then `guard`.